### PR TITLE
:sparkles: [Add] Coverage dir for unpack pod

### PR DESCRIPTION
**Pull request by varshaprasad96**
This PR adds the go test coverage variable to the
unpack pod. It also creates necessary volumes
if the GOCOVERDIR env var is set.

### CODE REVIEW

File: pkg/source/image.go [hunk 1]
```diff
@@ -139,28 +146,54 @@ func (i *Image) getDesiredPodApplyConfig(bundle *rukpakv1alpha2.BundleDeployment
...
 				WithImage(bundle.Spec.Source.Image.Ref).
 				WithCommand("/bin/unpack", "--bundle-dir", "/").
-				WithVolumeMounts(applyconfigurationcorev1.VolumeMount().
-					WithName("util").
-					WithMountPath("/bin"),
-				).
+				WithVolumeMounts(func() []*applyconfigurationcorev1.VolumeMountApplyConfiguration {
+					var volumeMounts []*applyconfigurationcorev1.VolumeMountApplyConfiguration
+					if gocoverdirEnv != "" {
+						volumeMounts = append(volumeMounts, applyconfigurationcorev1.VolumeMount().
```
Comment (varshaprasad96) [inline pkg/source/image.go:172]: A temporary fix to get e2e's running in operator controller!

File: pkg/source/image.go [hunk 2]
```diff
@@ -139,28 +146,54 @@ func (i *Image) getDesiredPodApplyConfig(bundle *rukpakv1alpha2.BundleDeployment
...
-					WithName("util").
-					WithMountPath("/util/bin"),
-				).
+				WithVolumeMounts(func() []*applyconfigurationcorev1.VolumeMountApplyConfiguration {
+					var volumeMounts []*applyconfigurationcorev1.VolumeMountApplyConfiguration
+					if gocoverdirEnv != "" {
+						volumeMounts = append(volumeMounts, applyconfigurationcorev1.VolumeMount().
+							WithName("test-coverage").
+							WithMountPath(gocoverdirEnv))
+					}
```
Comment (joelanford) [inline pkg/source/image.go:155]: I don't think we need coverage in this container, right? There are no instrumented binaries here.

File: pkg/source/image.go [hunk 3]
```diff
@@ -139,28 +146,54 @@ func (i *Image) getDesiredPodApplyConfig(bundle *rukpakv1alpha2.BundleDeployment
...
+					volumeMounts = append(volumeMounts, applyconfigurationcorev1.VolumeMount().
+						WithName("util").
+						WithMountPath("/util/bin"))
+					return volumeMounts
+				}()...).
+				WithEnv(applyconfigurationcorev1.EnvVar().WithName("GOCOVERDIR").WithValue(gocoverdirEnv)).
```
Comment (joelanford) [inline pkg/source/image.go:161]: Similar comment as above: this container doesn't need GOCOVERDIR set, right?

File: pkg/source/image.go [hunk 4]
```diff
@@ -139,28 +146,54 @@ func (i *Image) getDesiredPodApplyConfig(bundle *rukpakv1alpha2.BundleDeployment
...
+						volumeMounts = append(volumeMounts, applyconfigurationcorev1.VolumeMount().
+							WithName("test-coverage").
+							WithMountPath(gocoverdirEnv))
+					}
+					volumeMounts = append(volumeMounts, applyconfigurationcorev1.VolumeMount().
+						WithName("util").
+						WithMountPath("/bin"))
+					return volumeMounts
+				}()...).
+				WithEnv(applyconfigurationcorev1.EnvVar().WithName("GOCOVERDIR").WithValue(gocoverdirEnv)).
```
Comment (joelanford) [inline pkg/source/image.go:181]: Can we only set this if `gocoverdirEnv != ""`?

File: pkg/source/image.go (hunk 2 above)
Comment (varshaprasad96) [reply pkg/source/image.go]: That's right, init container doesn't need it. Reverted back.

**Review by everettraven — DISMISSED** [review]

**Review by everettraven — DISMISSED** [review]

**Review by joelanford — APPROVED** [review]

**Review by tmshort — DISMISSED** [review]
/lgtm

File: pkg/source/image.go (hunk 4 above)
Comment (varshaprasad96) [reply pkg/source/image.go:172]: Passing an empty configuration is causing errors. Tried various options, but looks like the only one is to create the entire container spec separately based on whether env var is present or not. 

Leaving it to be as is for now, if the env is not set it results in an empty string which doesn't cause a problem for now. Also this is just a band aid, and will replaced by direct registry client anyways.

**Review by tmshort — APPROVED** [review]

**Pull request merged**